libs/ledger/src/chain/main_chain.cpp

```diff
@@ -1797,6 +1797,13 @@ MainChain::IntBlockPtr MainChain::CreateGenesisBlock()
   return genesis;
 }
 
+bool MainChain::IsGenesisBlock(Block const &block)
+{
+  return block.block_number == 0 && block.previous_hash == chain::ZERO_HASH &&
```

Comment:
This function actually exists in the block - see suggestion below. Would be worth checking with @denis-trapeznikov - currently I believe I think it only checks with a zero hash to signal genesis. But I think that this can be updated now I think to add at least the genesis digest check here.

I would not make this change (altering `IsGenesis()` method) in this PR (for the release branch) but would update when we port this fix to `master`. Thoughts?

Reply:
Sounds good to me. Denis?